slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/JdbcMiscTest.scala

```diff
@@ -57,16 +57,26 @@ class JdbcMiscTest extends AsyncTest[JdbcTestDB] {
     }
     val t = TableQuery[T]
 
+    class U(tag: Tag) extends Table[Int](tag, "u".withUniquePostFix) {
+      def id = column[Int]("a")
+      def * = id
+    }
+    val u = TableQuery[U]
+
     val a1 = t.filter(_.id === 1)
     val a2 = t.filter(_.id === 2)
 
     seq(
-      t.schema.create,
+      (t.schema ++ u.schema).create,
       t ++= Seq(1, 2, 3),
       a1.result.map(_ shouldBe Seq(1)),
       a1.result.overrideStatements(a2.result.statements).map(_ shouldBe Seq(2)),
       a1.result.head.map(_ shouldBe 1),
-      a1.result.head.overrideStatements(a2.result.head.statements).map(_ shouldBe 2)
+      a1.result.head.overrideStatements(a2.result.head.statements).map(_ shouldBe 2),
+      (t += 4).overrideStatements(Seq(u.insertStatement)),
```

Comment:
Can you add some comments to the code explaining what the test is doing? Thanks.

Reply:
Ah, so the override is only for the "standard" part of the query (not counting the VALUES expression) and that is what `TableQuery[U].insertStatement` returns? Interesting.

I think I would find it less confusing if you just supplied a string instead of u.insertStatement, that way it's more focused on the behavior of overrideStatements and not on what insertStatement does, and I don't have to know what insertStatement does to understand the test.

Thoughts?

Reply:
I think I'll have to revert to `u.insertStatement`. The tests are run on different DB engines and each of them supports a slightly different syntax (think ```insert into `u` ``` vs. `insert into "u"`).

Reply:
Btw, `u.insertStatement` returns the whole statement (`insert into u (id) values (?)`). I updated the comment.

Reply:
What I'm confused about then is how the same statement works for inserting a single value and for inserting multiple rows. Not that has anything to do with your contribution, I'm just not clear how it works. Does ++= generate multiple insert statements? Or does the ? placeholder somehow support multiple rows?

Reply:
AFAIK, Slick translates `++=` to JDBC code similar to this: https://stackoverflow.com/a/3786127. So it "prepares" a statement from a string with placeholders, and then adds values from each row as a separate batch.

Reply:
Interesting